crates/vsock/src/main.rs

```diff
@@ -13,38 +13,83 @@ use std::{
     sync::{Arc, RwLock},
 };
 
-use clap::Parser;
+use clap::{Args, Parser};
 use log::{info, warn};
-use vhost::{vhost_user, vhost_user::Listener};
+use serde::Deserialize;
+use vhost::{
+    vhost_user,
+    vhost_user::{message::VhostUserConfig, Listener},
+};
 use vhost_user_backend::VhostUserDaemon;
 use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
 
 use crate::vhu_vsock::{Error, Result, VhostUserVsockBackend, VsockConfig};
 
-#[derive(Parser, Debug)]
-#[clap(version, about, long_about = None)]
-struct VsockArgs {
+#[derive(Args, Debug, Deserialize)]
+struct VsockSpec {
     /// Context identifier of the guest which uniquely identifies the device for its lifetime.
-    #[clap(long, default_value_t = 3)]
+    #[arg(long, default_value_t = 3, conflicts_with = "config")]
     guest_cid: u64,
 
     /// Unix socket to which a hypervisor connects to and sets up the control path with the device.
-    #[clap(long)]
+    #[arg(long, conflicts_with = "config")]
     socket: String,
 
     /// Unix socket to which a host-side application connects to.
-    #[clap(long)]
+    #[arg(long, conflicts_with = "config")]
     uds_path: String,
 }
 
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct VsockArgs {
+    #[command(flatten)]
+    spec: Option<VsockSpec>,
+
+    /// Load from a given configuration file
+    #[arg(short = 'f')]
```

Comment:
please, use only long options like for the other options

Reply:
This is now fixed.